language/e2e-tests/src/executor.rs

```diff
@@ -256,4 +266,41 @@ impl FakeExecutor {
         assert!(lcs::from_bytes::<NewBlockEvent>(event.event_data()).is_ok());
         self.apply_write_set(output.write_set());
     }
+
+    fn module(name: &str) -> ModuleId {
+        ModuleId::new(CORE_CODE_ADDRESS, Identifier::new(name).unwrap())
+    }
+
+    fn name(name: &str) -> Identifier {
+        Identifier::new(name).unwrap()
+    }
+
+    pub fn exec(
+        &mut self,
+        module_name: &str,
+        function_name: &str,
+        type_params: Vec<TypeTag>,
+        args: Vec<Value>,
+        sender: AccountAddress,
+    ) {
+        let write_set = {
+            let cost_table = zero_cost_schedule();
+            let mut cost_strategy = CostStrategy::system(&cost_table, GasUnits::new(100_000_000));
+            let vm = MoveVM::new();
```

Comment:
Is creating a fresh VM for every account creation expensive? I suppose we could add this to `self` if so.

Reply:
That is the main question in my mind as well! The main concern is if we were to introduce the writeset functionality in those tests we would need to think about the VM dropping strategy. That's why I did it in a fresh way. Plus the current transaction execution api creates a new MoveVM instance at every invocation. I'd love to add this to self as well if that's more efficient. Thoughts? cc @dariorussi